### PR TITLE
chore: allVariables tests for local and cloud sdk

### PR DESCRIPTION
**Pull request by nsalamad**

### CODE REVIEW

**Review by jsalaber — CHANGES_REQUESTED** [review]

File: harness/features/allVariables.local.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            await createClient(url, clientId, sdkKey, { baseURLOverride: `${mockServerUrl}/client/${clientId}` })
+            await wait(1000)
+        })
+
+        afterAll(() => {
+            nock.removeInterceptor(configInterceptor)
+        })
+
+        describeIf(capabilities.includes(Capabilities.local))(name, () => {
+            it('should return an empty object if client is not initialized', async () => {
```
Comment (ajwootto) [inline harness/features/allVariables.local.test.ts:47]: seems like this is more testing what happens when the SDK token is invalid?

Comment (nsalamad) [reply harness/features/allVariables.local.test.ts:47]: I didn't know how else to put the sdk in an uninitialized state. Got any suggestions?

File: harness/features/allVariables.local.test.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        beforeAll(async () => {
+            configInterceptor = scope
+                .get(`/client/${clientId}/config/v1/server/${sdkKey}.json`)
+            configInterceptor
+                .reply(200, config)
+                .persist()
+
+            url = getConnectionStringForProxy(name)
+            await createClient(url, clientId, sdkKey, { baseURLOverride: `${mockServerUrl}/client/${clientId}` })
+            await wait(1000)
```
[the remark this answers is not in the document]
Comment (nsalamad) [reply harness/features/allVariables.local.test.ts:39]: yup, will change this when the helper method gets added

File: harness/features/allVariables.local.test.ts (hunk 1 above)
Comment (ajwootto) [reply harness/features/allVariables.local.test.ts:47]: just make a new client then immediately call your method, that should do it

File: harness/features/allVariables.cloud.test.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                clientId,
+                sdkKey,
+                {
+                    baseURLOverride: `${mockServerUrl}/client/${clientId}`,
+                    enableCloudBucketing: true,
+                }
+            )
+        })
+
+        describeIf(capabilities.includes(Capabilities.cloud))(name, () => {
```
Comment (jsalaber) [inline harness/features/allVariables.cloud.test.ts:39]: could add a test for invalid user being passed

**Review by jsalaber — CHANGES_REQUESTED** [review]

File: harness/features/allVariables.cloud.test.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import { variables } from '../mockData'
+
+jest.setTimeout(10000)
+
+describe('allVariables Tests - Cloud', () => {
+    const scope = getServerScope()
+
+    forEachSDK((name: string) => {
+        const capabilities: string[] = SDKCapabilities[name]
+        const sdkKey = 'server_SDK_KEY'
```
Comment (elliotCamblor) [inline harness/features/allVariables.cloud.test.ts:22]: should update the sdk key to include clientId

File: harness/features/allVariables.cloud.test.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            it('should make a request to the variables endpoint with edgeDB param to true', async () => {
+                const clientId = uuidv4()
+                scope
+                    .post(`/client/${clientId}/v1/variables`)
+                    .query((queryObj) => {
+                        return queryObj.enableEdgeDB === 'true'
+                    })
+                    .reply(200, variables)
+
+                await createClient(
```
Comment (elliotCamblor) [inline harness/features/allVariables.cloud.test.ts:103]: use a new sdkKey

File: harness/features/allVariables.local.test.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+                const user = {
+                    user_id: 'test_user',
+                    email: 'user@gmail.com',
+                    age: '0',
+                    customData: { 'should-bucket': true }
+                }
+                const userResponse = await createUser(url, user)
+                const userLocation = userResponse.headers.get('Location')
+                const response = await callAllVariables(clientId, url, userLocation)
+                const { data: variablesMap } = await response.json()
```
Comment (elliotCamblor) [inline harness/features/allVariables.local.test.ts:93]: does this test need an assertion about the `entityType`? I think its supposed to return a plain object and not DVCVariables.

**Review by elliotCamblor — APPROVED** [review]

**Review by jsalaber — APPROVED** [review]

**Pull request merged**